packages/optimizer/lib/transformers/ApplyLayout.js

```diff
@@ -79,6 +80,13 @@ function apply(layout, width, height, node) {
     case 'container':
       // Do nothing here.
       break;
+    case 'fluid':
+      if (width.isSet) {
+        styles = `width:${width.numeral}${width.unit};`;
```

Comment:
When looking into it, it seems `width` isn't actually used or respected by fluid ads. Instead, there's a `width: 100% !important` set by https://github.com/ampproject/amphtml/blob/0320666d03db37c4e0e363b55f828aa807ae321c/extensions/amp-ad/0.1/amp-ad.css#L76-L78

Reply:
The Fluid docs [say](https://github.com/ampproject/amphtml/blob/main/extensions/amp-ad-network-doubleclick-impl/fluid.md):

> It will always occupy the maximum available width, and its height will be determined relative to that width.

But also:

> Note also that the width attribute is optional, and can be specified. When specified, the fluid creative will always occupy that width (unless used in conjunction with multi-size).

So it seem there is still a case where the `width` would be used?

Reply:
I don't believe it can be, due to the `!important` override?

Reply:
Since the selector is for `amp-ad[data-a4a-upgrade-type='amp-ad-network-doubleclick-impl'][height='fluid']`, is the `data-a4a-upgrade-type="amp-ad-network-doubleclick-impl"` attribute always added at runtime?

Reply:
Yup, doubleclick is the only thing that supports fluid, and that attr gets injected when `amp-ad` loads.

Reply:
Removed in d232ad24e5ab5a238843a5878df05b4bc39031dd